mhs/mhs/common/state/tests/test_dynamo_persistence_adaptor.py

```diff
@@ -30,146 +28,213 @@
+        response = await self.service.delete(TEST_INVALID_KEY)
 
         self.assertIsNone(response)
 
+    @patch.object(dynamo_persistence_adaptor.aioboto3, "resource")
     @async_test
-    async def test_delete_io_exception(self):
+    async def test_delete_io_exception(self, mock_boto3_resource):
+        """Test unhappy path for delete where an IO exception occurs."""
+        self.__configure_mocks(mock_boto3_resource)
         self.mock_table.delete_item.side_effect = TEST_SIDE_EFFECT
 
         with self.assertRaises(RecordDeletionError) as ex:
-            await self.service.delete(TEST_STATE_TABLE, TEST_INVALID_KEY)
+            await self.service.delete(TEST_INVALID_KEY)
 
         self.assertIs(ex.exception.__cause__, TEST_EXCEPTION)
 
     @staticmethod
-    def __setFutureResponse(method, response):
+    def __setFutureResponse(method, response) -> None:
+        """
+        Utility method to help setting expected responses on an async/await call.
+        :param method: The method to add the response to.
+        :param response: The response to be added to the method.
+        """
         future = asyncio.Future()
         future.set_result(response)
 
         method.return_value = future
+
+    def __configure_mocks(self, mock_boto3_resource):
+        """Configure the standard mocks to have mappings which can be used in the tests."""
+        mock = MagicMock()
+        self.mock_table = mock.Table.return_value
+        mock_boto3_resource.return_value = constant_context_manager(mock)
+
+
+@contextlib.asynccontextmanager
+async def constant_context_manager(mock: Mock):
+    """
+    Create a context manager which will provide a mock.
+    :param mock: The mock to provide.
+    :return: The mock required.
+    """
+    yield mock
+
+
+def single_loop_async_test(f):
+    """
+    A wrapper for asynchronous tests.
+    By default unittest will not wait for asynchronous tests to complete even if the async functions are awaited.
+    By annotating a test method with `@async_test` it will cause the test to wait for asynchronous activities
+    to complete. This will only use a single event loop for all tests with this annotation.
+    :param f: The coroutine which will be executed.
+    :return: a function which will execute an asynchronous unit test.
+    """
+
+    def wrapper(*args, **kwargs):
+        """
+        The wrapper function which will execute the provided coroutine within the shared event loop until it has
+        completed.
+        :param args: Arguments passed into the coroutine.
+        :param kwargs: Key word arguments passed into the coroutine.
+        :return: Only once the coroutine has completed.
+        """
+        coro = asyncio.coroutine(f)
+        future = coro(*args, **kwargs)
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(future)
+
+    return wrapper
+
+
+@unittest.skip("This has been configured for a specific test dynamo instance which may not exist.")
```

Comment:
So what is required to not skip these tests? Do we have a task covering unblocking them? Or should that be done as part of this task?

Reply:
These tests are just component integration tests to enable the verification the dynamodb code works with a real dynamodb. Unfortunately, this requires a real dynamodb and config to connect to it, and as component integration tests are not part of our test strategy, these have been ignored so as not to create test failures if their dependencies do not exist or have not been set up.